proposals/jib-cli-surface.md

```diff
@@ -0,0 +1,60 @@
+# Jib CLI Surface Specification
+
+Specification for a command structures exposed to users of the jib-cli
+
+## Usage
+`jib COMMAND [OPTIONS]`
+
+## Commands
+
+The jib cli exposes the following top level commands, mirroring what is available in the plugins
+```
+build        build to registry
+dockerBuild  build to docker daemon
+buildTar     build to a tar file
+```
+
+## Options
+
+### Required
+```
+-t  --target <image-ref>  a target image reference
+```
+
+### Optional
+```
+-b, --build-file <file>                    location of the build file (default <context>/jib.yaml)
+-c, --context <dir>                        location of the build context (default .)
+    --credHelper <registry>=<credHelper>   credential helper to use for a registry (repeatable)
```

Comment:
No simple way to pass a username and password pair if there's no credential helper or a Docker CLI? However, I don't know what's a conventional way to safely pass credentials for a CLI tool.

Anyways, `--allow-insecure` or `--insecure-registries=gcr.io,ghcr.io,...` sounds like a useful option. (Unfortunately, jib-core doesn't support per-registry configuration.)

Reply:
Added some more flags here to cover these